### PR TITLE
Simplify devicelab logic and fix tests

### LINKED ISSUE #53707: devicelab/test/run_test.dart is flaky
This a fresh reporting of #5901 - when it was closed it recommended opening a new issue to discuss if it resurfaced.

In doing a skip audit, I have found this is still broken.

Here is the associated skip:
https://github.com/flutter/flutter/blob/ad07c4041a966415ab96ebe4ee174ee43093c233/dev/devicelab/test/run_test.dart#L56

And here are some logs, it appears to either fail or hang  depending on the platform:

https://cirrus-ci.com/task/6551240146419712

https://cirrus-ci.com/task/5145431382425600

### PR DESCRIPTION
**Pull request by Hixie**
- fix https://github.com/flutter/flutter/issues/53707 by having the test not expect a timeout but instead actually look for the retry message
- simplify the `--task` option to only accept task names rather than also accepting paths
- remove some obsolete options that referred to the manifest which no longer seems to exist

### CODE REVIEW

File: dev/devicelab/bin/run.dart [hunk 1]
```diff
@@ -235,29 +234,12 @@ ArgParser createArgParser(List<String> taskNames) {
     ..addMultiOption(
       'task',
       abbr: 't',
-      help: 'Either:\n'
-          ' - the name of a task defined in manifest.yaml.\n'
-          '   Example: complex_layout__start_up.\n'
-          ' - the path to a Dart file corresponding to a task,\n'
-          '   which resides in bin/tasks.\n'
-          '   Example: bin/tasks/complex_layout__start_up.dart.\n'
+      help: 'Name of a Dart file in bin/tasks.\n'
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply dev/devicelab/bin/run.dart:237]: (I have no beef with removing and simplifying this, but I also don't know if the old behavior is used anywhere. Let's see if anything fails...)

File: dev/devicelab/test/run_test.dart [hunk 2]
```diff
@@ -12,45 +13,54 @@ import 'common.dart';
...
   const ProcessManager processManager = LocalProcessManager();
+  final String dart = path.absolute(
+    path.join('..', '..', 'bin', 'cache', 'dart-sdk', 'bin', 'dart'));
 
   group('run.dart script', () {
-    Future<ProcessResult> runScript(List<String> testNames,
+    // The tasks here refer to files in ../bin/tasks/*.dart
+
+    Future<ProcessResult> runScript(List<String> taskNames,
         [List<String> otherArgs = const <String>[]]) async {
```
Comment (goderbauer) [inline dev/devicelab/test/run_test.dart:23]: nit (unrelated to this change): formatting here is kinda strange.

**Review by goderbauer — APPROVED** [review]
Welcome to the project, new non-googler contributor! 😝

File: dev/devicelab/bin/run.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hixie) [reply dev/devicelab/bin/run.dart:237]: done

File: dev/devicelab/test/run_test.dart (hunk 2 above)
Comment (Hixie) [reply dev/devicelab/test/run_test.dart:23]: What would you prefer?

**Pull request merged**